File: tests/fixtures/retrieval/polyglot/go/sample.go
package polyglot

type Item struct {
	Price float64
	Qty   int
}

func ComputeTotal(items []Item) float64 {
	var total float64
	for _, it := range items {
		total += it.Price * float64(it.Qty)
	}
	return total
}

type OrderProcessor struct {
	TaxRate float64
}

func (o OrderProcessor) Process(items []Item) float64 {
	return ComputeTotal(items) * (1 + o.TaxRate)
}

File: tests/fixtures/retrieval/polyglot/python/sample.py
"""Synthetic polyglot fixture for retrieval tests."""


def compute_total(items):
    total = 0
    for item in items:
        total += item.get("price", 0) * item.get("qty", 1)
    return total


class OrderProcessor:
    def __init__(self, tax_rate=0.0):
        self.tax_rate = tax_rate

    def process(self, items):
        subtotal = compute_total(items)
        return subtotal * (1 + self.tax_rate)
